### DISCUSSION TITLE
How to decrypt Private Cookies value？

### DISCUSSION THREAD

**Discussion by zgangguoguo** [rwf2/Rocket#2245] [Questions] [answered]
`add_private` encrypts the content,how to decrypt ?

**Comment by wezm** [comment, marked as answer]
>The API for retrieving, adding, and removing private cookies is identical except that most methods are suffixed with _private. These methods are: [get_private](https://api.rocket.rs/v0.5-rc/rocket/http/struct.CookieJar.html#method.get_private), [get_pending], [add_private](https://api.rocket.rs/v0.5-rc/rocket/http/struct.CookieJar.html#method.add_private), and [remove_private](https://api.rocket.rs/v0.5-rc/rocket/http/struct.CookieJar.html#method.remove_private).

— https://rocket.rs/v0.5-rc/guide/requests/#private-cookies

**Reply by zgangguoguo** [reply]
> > The API for retrieving, adding, and removing private cookies is identical except that most methods are suffixed with _private. These methods are: [get_private](https://api.rocket.rs/v0.5-rc/rocket/http/struct.CookieJar.html#method.get_private), [get_pending], [add_private](https://api.rocket.rs/v0.5-rc/rocket/http/struct.CookieJar.html#method.add_private), and [remove_private](https://api.rocket.rs/v0.5-rc/rocket/http/struct.CookieJar.html#method.remove_private).
> 
> — https://rocket.rs/v0.5-rc/guide/requests/#private-cookies

I print the Cookie and I find the Value
![截屏2022-06-24上午9 17 02](https://user-images.githubusercontent.com/45198849/175440570-7e4f28a4-0e24-43f5-8837-5e28e110220e.png)
